### PR TITLE
Add active heading highlight and fix dead links for table of contents

### PR DESCRIPTION
**Pull request by Richie78321**
Addressing #128 and #242. 

Decided to add h5 and h6 to the autolinker, as I think we should follow the convention set by HTML for 6 depth levels of headers.

Still working on how to best implement styling change to avoid TOC overlap with guide content, which can happen quite often without contrived examples. I'm thinking that it might require some amount of either layout refactoring or some additional styling props to layouts..

### CODE REVIEW

**Comment by Richie78321** [comment]
Not sure why deploy failed but I can't see logs for Netlify unfortunately.

**Comment by varughese** [comment]
If you log into netlify with the pittcsc gmail (password in the csc drive somewhere) you can view the logs, I checked on my phone for a second and it looks like a graphql query is failing

File: src/components/table-of-contents.js [hunk 1]
```diff
@@ -9,7 +10,39 @@ const truncate = (word) => {
 }
 
 export default function TableOfContents({ headings }) {
+  const [activeHeading, updateActiveHeading] =
+      useState(headings.length > 0 ? headings[0] : null);
+
+  const checkForNewActiveHeading = throttle((e) => {
```
Comment (varughese) [inline src/components/table-of-contents.js:16]: Whats throttle do?

File: src/components/templates/guide-template.js [hunk 2]
```diff
@@ -51,6 +51,7 @@ export const pageQuery = graphql`
         related
       }
       headings {
+        id
```
Comment (varughese) [inline src/components/templates/guide-template.js:54]: I this is causing the netlify build to fail, not sure why

File: src/styles/components.scss [hunk 3]
```diff
@@ -222,6 +222,11 @@ header {
     margin-left: 50rem;
     @apply text-xs;
   }
+
+  .toc-heading-depth-6 {
```
Comment (varughese) [inline src/styles/components.scss:226]: Maybe we could use text overflow ellipsis to prevent the overflow problem?

**Review by varughese — COMMENTED** [review]
Looks good! Couple of questions 

- do we need to use underscore's throttle?
- is the id causing the gatsby build to fail on pages that don't have an id?

File: src/components/table-of-contents.js (hunk 1 above)
Comment (varughese) [reply src/components/table-of-contents.js:16]: If we need to use this don't we need to add underscore to the package json?

**Comment by varughese** [comment]
Also if you add 

close #128
close #242

to the PR description it will auto close the issues when this merges

File: src/components/table-of-contents.js (hunk 1 above)
Comment (Richie78321) [reply src/components/table-of-contents.js:16]: I was considering implementing it myself, but underscore's throttle is well-liked and it was already a dependency

Comment (Richie78321) [reply src/components/table-of-contents.js:16]: Check [here](https://underscorejs.org/docs/modules/throttle.html) for what it does. In a nutshell it prevents a function from running more frequently than the specified number of milliseconds (here 200)

Comment (Richie78321) [reply src/components/table-of-contents.js:16]: For the onscroll event, it is called soooooo many times so to avoid a slowdown I'm using the throttle

Comment (varughese) [reply src/components/table-of-contents.js:16]: word - you still need to add it to package json though: https://github.com/PittCSWiki/pittcswiki/blob/master/package.json#L11 Im guessing its a dependency of something else but its better to make that explicit in our package.json (imagine if one of our dependencies that uses underscore removes it, then we would be missing underscore)

File: src/components/templates/guide-template.js (hunk 2 above)
Comment (Richie78321) [reply src/components/templates/guide-template.js:54]: Yeah checked the logs and this is the problem. Looking into it now

File: src/styles/components.scss (hunk 3 above)
Comment (Richie78321) [reply src/styles/components.scss:226]: We already do use it, so we could either make it more strict or more strict depending on the depth (to account for the increased left margin for higher depths). Although with some reasonable styling it should be totally doable with styles. Will consider this tho.

**Comment by varughese** [comment]
richie and I noticed that id is in the heading - https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-remark/src/create-schema-customization.js - but the package.json had `gatsby-transformer-remark` at `^2.6` but we needed `^2.15` so updating that should fix the build error jawn

**Review by varughese — APPROVED** [review]

**Pull request merged**